src/sentry/api/endpoints/user_details.py

```diff
@@ -254,9 +254,10 @@ def delete(self, request: Request, user) -> Response:
         ]
 
         if remaining_org_ids:
-            OrganizationMember.objects.filter(
+            for member in OrganizationMember.objects.filter(
                 organization__in=remaining_org_ids, user_id=user.id
-            ).delete()
+            ):
+                member.delete()
```

Comment:
While this is more correct than before, we will need to change this further as we won't be able to reach orgmember from control silo (where this endpoint lives).